onnxruntime/core/providers/cpu/controlflow/if.cc

```diff
@@ -96,11 +96,19 @@ ONNX_CPU_OPERATOR_VERSIONED_KERNEL(If,
                                    If);
 
 // optional type is supported starting opset-16
+ONNX_CPU_OPERATOR_VERSIONED_KERNEL(If,
+                                   16, 18,
+                                   KernelDefBuilder()
+                                       .TypeConstraint("B", DataTypeImpl::GetTensorType<bool>())
+                                       .TypeConstraint("V", DataTypeImpl::AllTensorAndSequenceTensorAndOptionalTypes()),
+                                   If);
+
+// float 8 support was added.
 ONNX_CPU_OPERATOR_KERNEL(If,
-                         16,
+                         19,
                          KernelDefBuilder()
                              .TypeConstraint("B", DataTypeImpl::GetTensorType<bool>())
-                             .TypeConstraint("V", DataTypeImpl::AllTensorAndSequenceTensorAndOptionalTypes()),
+                             .TypeConstraint("V", DataTypeImpl::AllTensorAndSequenceTensorAndOptionalTypesIR9()),
```

Comment:
I don't see any tests to validate fp8 support for these control flow ops. Is this covered by onnx node tests?

Reply:
There is no test in onnx covering this. The code does not seem to use the type. I did not find any test with bfloat16 (float8 follows the same design). I assumed it was ok to not add one.